File: src/Moryx.AbstractionLayer.Products.Endpoints/Model/PartConnector.cs
// Copyright (c) 2023, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;
using Moryx.Serialization;

namespace Moryx.AbstractionLayer.Products.Endpoints
{
    [DataContract]
    public class PartConnector
    {
        [DataMember]
        public string Name { get; set; }

        [DataMember]
        public string DisplayName { get; set; }

        [DataMember]
        public string Type { get; set; }

        [DataMember]
        public bool IsCollection { get; set; }

        [DataMember]
        public PartModel[] Parts { get; set; }

        [DataMember]
        public Entry PropertyTemplates { get; set; }
    }
}

File: src/Moryx.Products.Web/src/app/api/models/part-connector.ts
/* tslint:disable */
/* eslint-disable */
import { Entry } from '@moryx/ngx-web-framework/entry-editor';
import { PartModel } from './part-model';
export interface PartConnector {
  displayName?: null | string;
  isCollection?: boolean;
  name?: null | string;
  parts?: null | Array<PartModel>;
  propertyTemplates?: Entry;
  type?: null | string;
}
